receiver/googlecloudmonitoringreceiver/internal/metrics_conversion.go

```diff
@@ -120,3 +121,96 @@ func (mb *MetricsBuilder) ConvertDeltaToMetrics(ts *monitoringpb.TimeSeries, m p
 
 	return m
 }
+
+func (mb *MetricsBuilder) ConvertDistributionToMetrics(metricKind metric.MetricDescriptor_MetricKind, ts *monitoringpb.TimeSeries, m pmetric.Metric) pmetric.Metric {
+	m.SetName(ts.GetMetric().GetType())
+	m.SetUnit(ts.GetUnit())
+	dist := m.SetEmptyHistogram()
+
+	switch metricKind {
+	case metric.MetricDescriptor_CUMULATIVE:
+		dist.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+	case metric.MetricDescriptor_DELTA:
+		dist.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+	default:
+		mb.logger.Warn("Unsupported metric kind:", zap.String("Metric", ts.GetMetric().GetType()))
+		return m
+	}
+
+	for _, point := range ts.GetPoints() {
+		dp := dist.DataPoints().AppendEmpty()
+
+		// Set timestamps if valid
+		if point.Interval.StartTime != nil && point.Interval.StartTime.IsValid() {
+			dp.SetStartTimestamp(pcommon.NewTimestampFromTime(point.Interval.StartTime.AsTime()))
+		}
+
+		if point.Interval.EndTime != nil && point.Interval.EndTime.IsValid() {
+			dp.SetTimestamp(pcommon.NewTimestampFromTime(point.Interval.EndTime.AsTime()))
+		} else {
+			mb.logger.Warn("EndTime is invalid for metric:", zap.String("Metric", ts.GetMetric().GetType()))
+		}
+
+		distValue := point.Value.GetDistributionValue()
+		if distValue == nil {
+			mb.logger.Warn("Distribution value is nil for metric:", zap.String("Metric", ts.GetMetric().GetType()))
```

Comment:
Should we do this check before we AppendEmpty to the datapoints so that we don't pass through empty distribution points?